tagmanager/ctags/powershell.c

```diff
@@ -0,0 +1,666 @@
+		case ':': token->type = TOKEN_COLON;				break;
+		case '{': token->type = TOKEN_OPEN_CURLY;			break;
+		case '}': token->type = TOKEN_CLOSE_CURLY;			break;
+		case '[': token->type = TOKEN_OPEN_SQUARE;			break;
+		case ']': token->type = TOKEN_CLOSE_SQUARE;			break;
+
+		case '=':
+		{
+			int d = fileGetc ();
+			if (d == '=' || d == '>')
+				token->type = TOKEN_OPERATOR;
+			else
+			{
+				fileUngetc (d);
+				token->type = TOKEN_EQUAL_SIGN;
+			}
+			break;
+		}
+
+		case '\'':
+		case '"':
+			token->type = TOKEN_STRING;
+			parseString (token->string, c);
+			token->lineNumber = getSourceLineNumber ();
+			token->filePosition = getInputFilePosition ();
+			break;
+
+		case '<':
+		{
+			int d = fileGetc ();
+			if (d == '#')
+			{
+				/* <# ... #> multiline comment */
+				do
+				{
+					c = skipToCharacter ('#');
+					if (c != EOF)
+					{
+						c = fileGetc ();
+						if (c == '>')
+							break;
+						else
+							fileUngetc (c);
+					}
+				} while (c != EOF);
+				goto getNextChar;
+			}
+			else
+			{
+				fileUngetc (d);
+				token->type = TOKEN_UNDEFINED;
+			}
+			break;
+		}
+
+		case '#': /* comment */
+			skipSingleComment ();
+			goto getNextChar;
+			break;
+
+		case '+':
+		case '-':
+		case '*':
+		case '/':
+		case '%':
+		{
+			int d = fileGetc ();
+			if (d != '=')
+				fileUngetc (d);
+			token->type = TOKEN_OPERATOR;
+			break;
+		}
+
+		case '$': /* variable start */
+		{
+			int d = fileGetc ();
+			if (! isIdentChar (d))
+			{
+				fileUngetc (d);
+				token->type = TOKEN_UNDEFINED;
+			}
+			else
+			{
+				parseIdentifier (token->string, d);
+				token->type = TOKEN_VARIABLE;
+			}
+			break;
+		}
+
+		default:
+			if (! isIdentChar (c))
+				token->type = TOKEN_UNDEFINED;
+			else
+			{
+				parseIdentifier (token->string, c);
+				if (isTokenFunction (token->string))
+					token->type = TOKEN_KEYWORD;
+				else
+					token->type = TOKEN_IDENTIFIER;
+			}
+			break;
+	}
+}
+
+static void enterScope (tokenInfo *const parentToken,
+						const vString *const extraScope,
+						const int parentKind);
+
+/* strip a possible PowerShell scope specification and convert it to accessType */
+static const char *const parsePowerShellScope (tokenInfo *const token)
+{
+	const char *access = ACCESS_UNDEFINED;
+	const char *const token_name = vStringValue (token->string);
+	const char *powershell_scope_end;
+
+	powershell_scope_end = strchr (token_name, ':');
+	if (powershell_scope_end)
+	{
+		int powershell_scope_len;
+		char *tag_name;
+		vString * powershell_scope = vStringNew ();
+
+		powershell_scope_len = powershell_scope_end - token_name;
+		/* cut the resulting scope string */
+		vStringNCopyS (powershell_scope, token_name, powershell_scope_len);
+		vStringTerminate (powershell_scope);
+		/* the rest after the ":" (colon) is the tag name */
+		tag_name = strdup (token_name + powershell_scope_len + 1);
+		/* update the token */
+		vStringClear (token->string);
+		vStringCopyS (token->string, tag_name);
```

Comment:
what about "just" doing:
```C
memmove(token->string->buffer + powershell_scope_len,
		token->string->buffer,
		token->string->len + 1 - powershell_scope_len);
token->string->len -= powershell_scope_len;
```
or similar?  This doesn't sound much more subtle to me, and avoids some temp copy.  Just a suggestion though.